cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/functional/BasicLatticesRedesign.kt

```diff
@@ -208,7 +203,7 @@ interface Lattice<T : Lattice.Element> {
 
 /** Implements a [Lattice] whose elements are the powerset of a given set of values. */
 class PowersetLattice<T>() : Lattice<PowersetLattice.Element<T>> {
-    override lateinit var elements: Set<Element<T>>
+    override lateinit var elements: LinkedHashSet<Element<T>>
 
     class Element<T>(expectedMaxSize: Int) : IdentitySet<T>(expectedMaxSize), Lattice.Element {
```

Comment:
Probably you want to change this line of code. We need a very fast and scalable implementation of a `Set` which keeps the order and compares based on the identity, not equality.